### PR TITLE
resolved some test issues

### LINKED ISSUE #279: fix flakey test
https://github.com/DaftDoris/newsdesk/runs/6661155370?check_suite_focus=true#step:4:350

### PR DESCRIPTION
**Pull request by bhavinsen**
Fixed #279

### CODE REVIEW

**Review by chrisns — CHANGES_REQUESTED** [review]

File: src/store/item.ts [hunk 1]
```diff
@@ -29,12 +29,12 @@ export const useItemStore = defineStore("item", {
...
 
       this.itemList.push(item)
-      return this.saveData(podcastname, docname)
+      this.saveData(podcastname, docname)
     },
 
     async updateSlotItem(item: [], podcastname: string, docname: string) {
       this.itemList = item
-      return this.saveData(podcastname, docname)
+      this.saveData(podcastname, docname)
```
Comment (chrisns) [inline src/store/item.ts:37]: without the return this is fire forget, you've no ability to `await updateSlotItem`

File: src/store/item.ts [hunk 2]
```diff
@@ -29,12 +29,12 @@ export const useItemStore = defineStore("item", {
       const item: Item = { ...params, id }
 
       this.itemList.push(item)
-      return this.saveData(podcastname, docname)
+      this.saveData(podcastname, docname)
```
Comment (chrisns) [inline src/store/item.ts:32]: without the return this is fire forget, you've no ability to `await addItem`

File: src/store/item.ts [hunk 3]
```diff
@@ -51,16 +51,22 @@ export const useItemStore = defineStore("item", {
...
+      // eslint-disable-next-line no-useless-catch
+      try {
+        setDoc(docRef, {
+          items: this.itemList,
+          slotTitles: this.slotTitleList,
+        })
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (e: any) {
+        throw e
+      }
```
Comment (chrisns) [inline src/store/item.ts:63]: not sure why you need the try/catch since you're not doing anything in the catch

File: src/store/item.ts [hunk 4]
```diff
@@ -51,16 +51,22 @@ export const useItemStore = defineStore("item", {
 
     async saveData(podcastname: string, docname: string) {
       const docRef = doc(collection(db, podcastname), docname)
-      return setDoc(docRef, {
-        items: this.itemList,
-        slotTitles: this.slotTitleList,
-      })
+      // eslint-disable-next-line no-useless-catch
+      try {
+        setDoc(docRef, {
```
Comment (chrisns) [inline src/store/item.ts:56]:
```suggestion
        await setDoc(docRef, {
```

File: src/store/itemShare.ts [hunk 5]
```diff
@@ -42,9 +42,11 @@ export const useShareStore = defineStore("share", {
           })
           // eslint-disable-next-line @typescript-eslint/no-explicit-any
         } catch (e: any) {
-          // create doc if it doesn't exist
           if (e.code === "not-found" && e.name === "FirebaseError")
-            await setDoc(docRef, { items: arrayUnion(item.text) })
+            try {
+              setDoc(docRef, { items: arrayUnion(item.text) })
+              // eslint-disable-next-line no-empty
+            } catch (err: unknown) {}
```
Comment (chrisns) [inline src/store/itemShare.ts:49]: not sure why you need the try/catch since you're not doing anything in the catch
you're also not `await` on the setDoc

**Review by chrisns — CHANGES_REQUESTED** [review]

**Review by chrisns — APPROVED** [review]

File: src/store/item.ts (hunk 1 above)
Comment (bhavinsen) [reply src/store/item.ts]: I have added `return`

File: src/store/item.ts (hunk 2 above)
Comment (bhavinsen) [reply src/store/item.ts]: I have added `return`

File: src/store/item.ts (hunk 4 above)
Comment (bhavinsen) [reply src/store/item.ts]: I have added `await`

File: src/store/itemShare.ts (hunk 5 above)
Comment (bhavinsen) [reply src/store/itemShare.ts]: I have updated this code.
please refer -> https://i.imgur.com/kPUWcBx.png

**Review by chrisns — APPROVED** [review]

**Pull request merged**